app/models/miq_provision_virt_workflow.rb

```diff
@@ -578,25 +578,17 @@ def allowed_template_condition
 
   def source_vm_rbac_filter(vms, condition = nil)
     filter_id = get_value(@values[:vm_filter]).to_i
-    search_options = {:user => @requester}
+    search_options = {:user => @requester, :class => VmOrTemplate, :results_format => :objects}
     search_options[:conditions] = condition unless condition.blank?
     template_msg =  "User: <#{@requester.userid}>"
     template_msg += " Role: <#{@requester.current_group.nil? ? "none" : @requester.current_group.miq_user_role.name}>  Group: <#{@requester.current_group.nil? ? "none" : @requester.current_group.description}>"
     template_msg += "  VM Filter: <#{@values[:vm_filter].inspect}>"
     template_msg += "  Passing inital template IDs: <#{vms.collect(&:id).inspect}>" unless vms.blank?
     _log.info "Checking for allowed templates for #{template_msg}"
-    if filter_id.zero?
-      if vms.empty?
-        Rbac.search(search_options.merge(:class => VmOrTemplate, :results_format => :objects)).first
-      else
-        Rbac.filtered(vms, search_options.merge(:class => VmOrTemplate))
-      end
+    if vms.empty?
+      MiqSearch.search(filter_id, search_options).first
     else
-      if vms.empty?
-        MiqSearch.find(filter_id).search([], search_options.merge(:results_format => :objects)).first
-      else
-        MiqSearch.find(filter_id).filtered(vms, search_options)
```

Comment:
There are 4 cases of code calling into `Rbac.search` (directly or indirectly)

1. `:class` is required
2. Code could derived `:class` from `vms.first.class`. It is sending anyway.
3. filter is defining `:class`
4. filter id defining `:class`, but doesn't need to.

I'm still leaning towards just passing it in and just making our code simpler.

As said before, I'd prefer all this code read:

```ruby
MiqSearch.filtered(filter_id, vms.presence, search_options.merge(:result_format => :objects, :class => VmOrTemplate)
```

but can only do that once filtered returns values for `targets.nil?`